File: source/old_cpp_src/_1_Two_Sum.cpp
#include<vector>

class Solution
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        map<int, int> a;
        vector<int>b(2, -1);
        for (int i = 0; i < nums.size(); i++)
        {
            /* code */
            if (a.count(target - nums[i]) > 0)
            {
                /* code */
                b[0] = a[target - nums[i]];
                b[1] = i;
                break;
            }
            a[nums[i]] = i;
            
        }
        return b;
        
    }
};

File: source/Go_LeetCode/1_TwoSum.go
package Go_LeetCode

func twoSum(nums []int, target int) []int {
	// map的key是某个之前的数， value是这个数出现的位置
	numMap := make(map[int]int)
	for i, num := range nums {
		if index, ok := numMap[target-num]; ok {
			return []int{index, i}
		}
		numMap[num] = i
	}
	return []int{-1, -1}
}
